pmd-core/src/main/java/net/sourceforge/pmd/RuleSetParser.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+import net.sourceforge.pmd.util.ResourceLoader;
+
+/**
+ * Configurable ruleset parser. Note that this replaces the API of {@link RulesetsFactoryUtils}
+ * and {@link RuleSetFactory}. This can be configured using a fluent
+ * API, see eg {@link #warnDeprecated(boolean)}. To create a list of
+ * rulesets, use {@link #parseFromResource(String)}.
+ */
+public final class RuleSetParser {
```

Comment:
Hm... part of this class is a builder not a parser....

Maybe we can start with phrasing what this class does: Loads rulesets from files (or other resources). The way, the rulesets are loaded, can be configured - hence configurable.

Not sure, if this is more a RuleSetBuilder, RuleSetLoader or RuleSetParser (well in PMD6 the impl of the parser is actually in RuleSetFactory, but that's only for now).

Since I came across RuleSetWriter... maybe this is a RuleSetReader? (Although this would make one think as a java.io.Reader, which works a bit different)

Reply:
RulesetLoader sounds good! I would avoid RulesetReader, sounds like something that uses a ruleset, not something that creates it. I think we could rename RulesetFactory to RulesetParser eventually